cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -123,6 +134,16 @@ public AppFabricServer(CConfiguration configuration, DiscoveryService discoveryS
     this.pluginService = pluginService;
     this.privilegesFetcherProxyService = privilegesFetcherProxyService;
     this.routeStore = routeStore;
+    this.sslEnabled = configuration.getBoolean(Constants.Security.AppFabric.SSL_ENABLED);
```

Comment:
We should probably have two to start with. One for SSL that terminates at the router, so all the external entities can use it and one for all of our internal entities. While minimal SSL does add an overhead so customers may want to have SSL only for connecting to router and none inside the cluster.